crates/evm/src/fuzz/mod.rs

```diff
@@ -101,76 +125,49 @@ impl<'a> FuzzedExecutor<'a> {
         let strat = proptest::strategy::Union::new_weighted(weights);
         debug!(func = ?func.name, should_fail, "fuzzing");
         let run_result = self.runner.clone().run(&strat, |calldata| {
-            let call = self
-                .executor
-                .call_raw(self.sender, address, calldata.0.clone(), 0.into())
-                .map_err(|_| TestCaseError::fail(FuzzError::FailedContractCall))?;
-            let state_changeset = call
-                .state_changeset
-                .as_ref()
-                .ok_or_else(|| TestCaseError::fail(FuzzError::EmptyChangeset))?;
-
-            // Build fuzzer state
-            collect_state_from_call(
-                &call.logs,
-                state_changeset,
-                state.clone(),
-                &self.config.dictionary,
-            );
-
-            // When assume cheat code is triggered return a special string "FOUNDRY::ASSUME"
-            if call.result.as_ref() == ASSUME_MAGIC_RETURN_CODE {
-                return Err(TestCaseError::reject(FuzzError::AssumeReject))
-            }
-
-            let success = self.executor.is_success(
-                address,
-                call.reverted,
-                state_changeset.clone(),
-                should_fail,
-            );
-
-            if success {
-                let mut first_case = first_case.borrow_mut();
-                if first_case.is_none() {
-                    first_case.replace(FuzzCase {
-                        calldata,
-                        gas: call.gas_used,
-                        stipend: call.stipend,
-                    });
+            let fuzz_res = self.single_fuzz(&state, address, should_fail, calldata)?;
+
+            match fuzz_res {
+                FuzzOutcome::Case(case) => {
+                    let mut first_case = first_case.borrow_mut();
+                    if first_case.is_none() {
+                        first_case.replace(case.case);
+                    }
+                    gas_by_case.borrow_mut().push((case.gas_used, case.stipend));
+
+                    traces.replace(case.traces);
+
+                    if let Some(prev) = coverage.take() {
+                        // Safety: If `Option::or` evaluates to `Some`, then `call.coverage` must
+                        // necessarily also be `Some`
+                        coverage.replace(Some(prev.merge(case.coverage.unwrap())));
+                    } else {
+                        coverage.replace(case.coverage);
+                    }
+
+                    Ok(())
```

Comment:
After this refactor (to be able to run only one fuzz if debug is on), it seems like the first rejection does not stop the test runner. Is that something related to `proptest` ?

Reply:
Right, I believe this might be something proptest related probably—it won't necessarily stop at the first rejection because it's looking for a counterexample and "shrinking" it—which means there might be something inefficient happening now. Maybe you could try flamegraphing this to see where it's taking so long? Also, here's the [proptest book](https://proptest-rs.github.io/proptest/intro.html)

Reply:
@Evalir it was working as intended, I was just not using the `--release` build lmao